### PR TITLE
Api spec

### PR DESCRIPTION
**Pull request by hgs-msmith**
Defining STAC as as an extension of WFS3 core, now defined at: https://github.com/opengeospatial/WFS_FES/blob/master/openapi.yaml
Includes two files:
* fragment that includes only the STAC elements to be added to WFS3 core
* complete API, union of WFS3 core and STAC

### CODE REVIEW

File: api-spec/WFS3core+STAC.yaml [hunk 1]
```diff
@@ -0,0 +1,631 @@
+openapi: 3.0.1
+info:
+  title: STAC API conforming to the OGC Web Feature Service standard
+  version: 0.0.1
+  description: >-
+    This is a sample OpenAPI definition that conforms to the OGC Web Feature
+    Service specification (conformance classes: "Core", "GeoJSON", "HTML" and
+    "OpenAPI 3.0").
```
Comment (cholmes) [inline api-spec/WFS3core+STAC.yaml]: Would be good to add in the fact that this also implements the STAC specification. Indeed the description should be more like 'This is a sample OpenAPI definition of a SpatioTemporal Asset Catalog API that also conforms to the OGC ...'

File: api-spec/WFS3core+STAC.yaml [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+      tags:
+        - STAC
+      requestBody:
+        content:
+          'application/json':      
+            schema:
+              $ref: '#/components/schemas/searchBody'
+      responses:
+        '200':
+          description: A feature.
```
Comment (cholmes) [inline api-spec/WFS3core+STAC.yaml]: Should add the STAC Item response here, the geojson as an example. The ones in the spec.yaml and wfs-stac yaml are up to date with the latest.

File: api-spec/WFS3core+STAC.yaml [hunk 3]
```diff
@@ -0,0 +1,631 @@
...
+                type: string
+# START STAC-specific operations                
+  /search/stac:
+    post:
+      summary: >-
+        retrieve items matching filters.
+      operationId: searchSTAC
+      tags:
+        - STAC
+      requestBody:
```
Comment (cholmes) [inline api-spec/WFS3core+STAC.yaml]: Can we specify the filters here in some way? Even if it's just an example that includes the two main filters.

**Comment by hgs-msmith** [comment]
Added STAC-specific comments
Added additional itemCollection types
Added /search/stac example

**Comment by hgs-msmith** [comment]
I will work up a STAC-only example next.

**Pull request merged**